### PR TITLE
ui: Add `stats-card` component, integrate more `@hashicorp/structure-icons`

### PR DESCRIPTION
**Pull request by johncowen**
1. Rebuild the `heathchecked-resource` component to use `base/stats-card` now we can select and copy values within it. Previously we'd used some more complicated absolute positioning CSS to ensure IP addresses could be selected and copied to the clipboard, since https://github.com/hashicorp/consul/pull/5770 we no longer need to do this.
2. As the above rebuild of this component came with new `@hashicorp/structure-icons`, we also swapped out 'most' of the other areas where we were using these new icons (the green tick, orange warning and red cross), plus any icons that were effected by the new icon placeholders (in the intention listing filter control bar).
3. We begin to remove more and more of the project specific icons (now replaced by the shared ones)

### CODE REVIEW

File: ui-v2/app/templates/components/healthchecked-resource.hbs [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
-    {{/each}}
-    {{#if (gt healthy.length 0)}}
-        <li class="passing">
-            <a href={{href}}>
-                <strong>{{healthy.length}}</strong>
-                <span>other passing {{pluralize healthy.length 'check' without-count=true}}</span>
-            </a>
-        </li>
+{{#stats-card}}
+  {{#block-slot 'icon'}}{{#if false}}<span data-tooltip="Leader">Leader</span>{{/if}}{{/block-slot}}
```
Comment (johncowen) [inline ui-v2/app/templates/components/healthchecked-resource.hbs]: P.S. This also adds a visible marker for being able to show which node is the leader. This is currently purposefully set to `false` until its 100% clear how we are going to get the information required for this from the API.

See:

https://github.com/hashicorp/consul/issues/5073
https://github.com/hashicorp/consul/pull/5455

https://www.consul.io/api/status.html#get-raft-leader

**Comment by johncowen** [comment]
@opihana and @jnwright I just remember I added the number of passing checks for the healthy nodes here with tooltip on rollover):

Screenshot 2019-06-26 at 14 08 17

I remember we had a feature request to add the number of services on the nodes on this screen (I think it was from @cbeckhashicorp ).

I almost asked if we should change this to the number of services rather than the number of passing checks, but then I thought that all the other information on this page uses these icons to signify *checks* not *services*, so I'm not sure if doing that might make things confusing? That fact that the icon we use here is a green tick means 'this is to signify a passing check' to me personally also.

Let me know what you think!

**Comment by opihana** [comment]
@johncowen Yeah, for now, it seems _more_ confusing to change the counts to services on the Nodes page, but I can keep that issue in mind.

File: ui-v2/tests/integration/components/stats-card-test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  integration: true,
+});
+
+test('it renders', function(assert) {
+  // Set any properties with this.set('myProperty', 'value');
+  // Handle any actions with this.on('myAction', function(val) { ... });
+
+  this.render(hbs`{{stats-card}}`);
+
+  assert.equal(
```
Comment (gregone) [inline ui-v2/tests/integration/components/stats-card-test.js]: maybe add actual tests?

Comment (johncowen) [reply ui-v2/tests/integration/components/stats-card-test.js]: Added!

File: ui-v2/app/templates/components/healthchecked-resource.hbs [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
+        <a href={{href}} class="passing">
+          <strong data-tooltip={{concat healthy.length ' other passing ' (pluralize healthy.length 'check' without-count=true)}}></strong>
+          <span>{{healthy.length}} other passing {{pluralize healthy.length 'check' without-count=true}}</span>
+        </a>
+      </li>
+  {{/if}}
+    </ul>
+  {{/if}}
+  {{/block-slot}}
+{{/stats-card}}
```
Comment (gregone) [inline ui-v2/app/templates/components/healthchecked-resource.hbs]: nit: eol

File: ui-v2/app/styles/components/modal-dialog/skin.scss [hunk 4]
```diff
@@ -21,8 +21,9 @@
   border-bottom-width: 1px;
 }
 %modal-window.warning > header {
-  @extend %with-warning;
-  text-indent: 20px;
+  /*TODO: warning isn't being used anywhere right now, but we should have modals */
```
Comment (gregone) [inline ui-v2/app/styles/components/modal-dialog/skin.scss]: errrr maybe remove this if unused

File: ui-v2/app/templates/components/healthchecked-resource.hbs (hunk 3 above)
Comment (johncowen) [reply ui-v2/app/templates/components/healthchecked-resource.hbs]: Added!

File: ui-v2/app/styles/components/modal-dialog/skin.scss (hunk 4 above)
Comment (johncowen) [reply ui-v2/app/styles/components/modal-dialog/skin.scss]: Removed!

**Pull request merged**